### DISCUSSION TITLE
is it possible to mark a formula as platform-independent?

### DISCUSSION THREAD

**Discussion by aryairani** [Homebrew/discussions#1640] [Writing Formulae/Casks] [answered]
So that the `brew tap-new` templates don't generate separate bottles for linux and mac?  In my case, the bottle is just a bunch of static html files that are built by node.js; the result is the same regardless of which platform it's built with.

**Comment by carlocab** [comment, marked as answer]
If the bottle has identical checksums across platforms, then the bottle commit will show `all` as the bottle tag. [Here's an example.](https://github.com/Homebrew/homebrew-core/blob/5490f05fd1842c08a11a50cdb19cf9576f73abeb/Formula/git-gui.rb#L10-L12)

**Reply by aryairani** [reply]
Ah, thanks!  Is that something the "brew pr-pull" workflow does automatically?

**Reply by carlocab** [reply]
Yes.

**Reply by aryairani** [reply]
Hm, it seems like it left them as two artifacts, even though the only difference between them is that the `INSTALL_RECEIPT.json` lists a different build platform. Is that expected?

**Reply by carlocab** [reply]
Yes. You'll want to pass `--only-json-tab` to `brew bottle` (or `brew test-bot`) to avoid sticking the tab (the install receipt) in the bottle tarball. Otherwise identical checksums is impossible. (You might also need the `--json` flag. `brew help {bottle,test-bot}` should tell you everything you need.)

**Reply by aryairani** [reply]
Hmm, I still got a different hash between mac versions and linux version even though the contents of the archive were the same.  Maybe some difference between the archiver on mac vs linux?

**Reply by carlocab** [reply]
Possibly. Can you show the output of `diffoscope` on the bottle tarballs? We've done a bit of work to minimise the diff between different macOS versions, but we haven't quite haven't seen taken a careful look at Linux vs macOS diffs. (By we I actually mean other Homebrew maintainers. I don't mean to take credit for them here.)

**Reply by aryairani** [reply]
Oooh, neat trick!  It looks like the Linux version has `go+w` permissions on a few things that the macOS versions don't.

![image](https://user-images.githubusercontent.com/538571/121273633-a68f7080-c896-11eb-9a82-2d46e1aa3baa.png)

My formula is [here](https://github.com/unisonweb/homebrew-unison-debug/blob/40697e52a796600b003b460b99b7fbcea402f082/Formula/ucm-codebase-ui.rb).

**Reply by carlocab** [reply]
Interesting. This one's above my pay grade, so I'll ping @Bo98 here. Hopefully, they have time to take a look 🙂

**Reply by aryairani** [reply]
Thanks for your help!

**Reply by carlocab** [reply]
One thing you can do here is to do `chmod` in the `install` method to try to make the permissions uniform across platforms. This is consistent with what we do in some formulae to try to make the bottle uniform across macOS versions, so it's not too surprising to have to do something similar to get a platform-independent bottle.

**Reply by Bo98** [reply]
Checksum matching between macOS and Linux is indeed a goal - just not tested much since linuxbrew-core have opted to not use `all` bottles for now.

Considering it is affecting `.brew`, this does seem like a bug. I'll investigate this.

What `umask` do you run (if changed)?
